File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.py
n,m=map(int,input().split())
num=[0]*(n+1)
for i in range(1,n+1):
    num[i]=i
for i in range(m):
    j,k=map(int,input().split())
    for t in range((k-j+1)//2):
        d=num[j+t]
        num[j+t]=num[k-t]
        num[k-t]=d
for i in range(1,n+1):
    print(num[i],end=' ')

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.cc
#include <iostream>
#include <vector>
using namespace std;

int main(void) {
    int n, m, a, b, t;
    cin >> n >> m;
    int num[n+1] = {0,};
    for (int i = 1; i < n + 1; i++)
        num[i] = i;
    for (int i = 0; i < m; i++) {
        cin >> a >> b;
        for (int j = 0; j < (b - a + 1) / 2; j++) {
            t = num[a + j];
            num[a + j] = num[b - j];
            num[b - j] = t;
        }
    }
    for (int i = 1; i < n + 1; i++)
        cout << num[i] << ' ';
    return 0;
}